dash.js

```diff
@@ -699,7 +699,20 @@ const MyDash = new Lang.Class({
     _redisplay: function() {
         let favorites = AppFavorites.getAppFavorites().getFavoriteMap();
 
-        let running = this._appSystem.get_running();
+        let running;
+        if (this._dtdSettings.get_boolean('isolate-workspaces')) {
+            running = this._appSystem.get_running().filter(function(_app) {
```

Comment:
Can't we reuse the `appIcons.getInterestingWindows()` function and filter for `windows!== null`?

Like:

```
let running = this._appSystem.get_running();

if (this._dtdSettings.get_boolean('isolate-workspaces')) {
    running = running.filter(function (.... ) {
           return appIcons.getInterestingWindows !== null;
     })
}